NewApp/ngsa-csharp/Ngsa.App/Controllers/ActorsController.cs

```diff
@@ -39,29 +35,55 @@ public ActorsController(ILogger<ActorsController> logger)
         [HttpGet]
         public async Task<IActionResult> GetActorsAsync([FromQuery] ActorQueryParameters actorQueryParameters)
         {
+            NgsaLog nLogger = Logger.GetLogger(nameof(GetActorsAsync), HttpContext).EnrichLog();
+            nLogger.LogInformation("Web Request");
+
             if (actorQueryParameters == null)
             {
                 throw new ArgumentNullException(nameof(actorQueryParameters));
             }
 
+            List<Middleware.Validation.ValidationError> list = actorQueryParameters.Validate();
+
+            if (list.Count > 0)
+            {
+                nLogger.Data.Clear();
+                nLogger.EventId = new EventId((int)HttpStatusCode.BadRequest, HttpStatusCode.BadRequest.ToString());
+                nLogger.LogWarning($"Invalid query string");
+
+                return ResultHandler.CreateResult(list, Request.Path.ToString() + (Request.QueryString.HasValue ? Request.QueryString.ToString() : string.Empty));
+            }
+
             return await DataService.Read<List<Actor>>(Request).ConfigureAwait(false);
         }
 
         /// <summary>
         /// Returns a single JSON Actor by actorId
         /// </summary>
-        /// <param name="actorIdParameter">The actorId</param>
+        /// <param name="actorId">The actorId</param>
         /// <response code="404">actorId not found</response>
         /// <returns>IActionResult</returns>
         [HttpGet("{actorId}")]
-        public async Task<IActionResult> GetActorByIdAsync([FromRoute] ActorIdParameter actorIdParameter)
+        public async Task<IActionResult> GetActorByIdAsync([FromRoute] string actorId)
         {
-            if (actorIdParameter == null)
+            NgsaLog nLogger = Logger.GetLogger(nameof(GetActorByIdAsync), HttpContext).EnrichLog();
+            nLogger.LogInformation("Web Request");
+
+            if (string.IsNullOrEmpty(actorId))
```

Comment:
Should we use IsNullOrWhitespace since it checks all 3 conditions?

Reply:
good catch - changed